hs_core/management/commands/list_resources.py

```diff
@@ -0,0 +1,118 @@
+from hs_core.hydroshare.utils import get_resource_by_shortkey
+
+
+def has_subfolders(resource):
+    for f in resource.files.all():
+        if '/' in f.short_path:
+            return True
+    return False
+
+
+def measure_resource(short_id):
+    """ Print size and sharing status of a resource """
+
+    try:
+        res = BaseResource.objects.get(short_id=short_id)
+    except BaseResource.DoesNotExist:
+        print("{} does not exist".format(short_id))
+
+    resource = res.get_content_model()
+    assert resource, (res, res.content_model)
+
+    istorage = resource.get_irods_storage()
+    if resource.raccess.public:
+        status = "public"
+    elif resource.raccess.discoverable:
+        status = "discoverable"
+    else:
+        status = "private"
+
+    if istorage.exists(resource.file_path):
+        print("{} {} {} {} {} {}".format(resource.size, short_id, status, resource.storage_type,
+                                         resource.resource_type, resource.title))
+    else:
+        print("{} {} {} {} {} {} NO IRODS FILES".format('-', short_id, status,
+                                                        resource.storage_type,
+                                                        resource.resource_type,
+                                                        resource.title))
+
+
+class Command(BaseCommand):
+    help = "Print size information."
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's: none does nothing.
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+        # Named (optional) arguments
+        parser.add_argument(
+            '--log',
+            action='store_true',  # True for presence, False for absence
+            dest='log',           # value is options['log']
+            help='log errors to system log',
+        )
+
+        parser.add_argument(
+            '--type',
+            dest='type',
+            help='limit to resources of a particular type'
+        )
+
+        parser.add_argument(
+            '--storage',
+            dest='storage',
+            help='limit to specific storage medium (local, user, federated)'
+        )
+
+        parser.add_argument(
+            '--access',
+            dest='access',
+            help='limit to specific access class (public, discoverable, private)'
+        )
+
+        parser.add_argument(
+            '--has_subfolders',
+            action='store_true',  # True for presence, False for absence
+            dest='has_subfolders',  # value is options['has_subfolders']
+            help='limit to resources with subfolders',
+        )
+
+    def handle(self, *args, **options):
+        if len(options['resource_ids']) > 0:  # an array of resource short_id to check.
+            for rid in options['resource_ids']:
+                resource = get_resource_by_shortkey(rid)
+                if (options['type'] is None or resource.resource_type == options['type']) and \
```

Comment:
This mammoth if block looks to be the same as the one following.... maybe put it in a method and reuse?